src/dataworker/DataworkerUtils.ts

```diff
@@ -89,7 +89,7 @@ export function blockRangesAreInvalidForSpokeClients(
     }
 
     const clientLastBlockQueried =
-      spokePoolClients[chainId].eventSearchConfig.toBlock ?? spokePoolClients[chainId].latestBlockNumber;
+      spokePoolClients[chainId].eventSearchConfig.toBlock ?? spokePoolClients[chainId].latestBlockSearched;
```

Comment:
We unset `toBlock` at the end of the `SpokePoolClient.update()` now, so it will never be defined outside of that context.

```suggestion
      spokePoolClients[chainId].latestBlockSearched;
```